src/build.py

```diff
@@ -129,39 +135,42 @@ def fix_comments(str):
 
 # Print concatenated output
 # -------------------------
-print("/*")
+sys.stdout.write("/*" + os.linesep)
```

Comment:
@Hejsil ?

Reply:
Sorry, I have no idea on this one. Windows and Python oddities are not something i deal with often.

Reply:
The problem is with print on Windows with Python 2.x (I have 2.7). For some reason, it always uses LF instead of CRLF. It is silly that we have to work around this kind of thing, which is (apparently) resolved in Python 3.x, but too many programs still rely on 2.x. Would you be willing to make the script only work with Python 3.x? Most developers should have 3.x on their systems nowadays anyway, and it's simple to install if they don't have it.

Reply:
>For some reason, it always uses LF instead of CRLF.

Hm, but that's exactly the behavior we require, don't we?

>Would you be willing to make the script only work with Python 3.x?

Yeah, that's also an option I'm totally OK with.

Reply:
> Hm, but that's exactly the behavior we require, don't we?

No, as git will fail to convert line endings if it detects the use of mixed line endings.

Reply:
>No, as git will fail to convert line endings if it detects the use of mixed line endings.

Pardon my ignorance, this seems to be the misunderstanding I'm failing to see. Where **exactly** (code line in this repo master branch) does the mixed encoding come from?

Reply:
> Pardon my ignorance, this seems to be the misunderstanding I'm failing to see. Where **exactly** (code line in this repo master branch) does the mixed encoding come from?

`print()` on Windows in Python 2.x (or at least in 2.7) appends `\n` to the end of the string you pass it. Git on Windows converts all line-endings to `\r\n` when you checkout, at least by default. It will then convert all line-endings back to `\n` when committing. However, it seems to fail if it detects multiple different line-endings; At least, this is the case with nuklear.h. This results in nuklear.h being committed with a mixture of `\r\n` and `\n`. Python 3.x has a fix for this, and properly outputs the line-endings that you would expect for your OS.

Reply:
>However, it seems to fail if it detects multiple different line-endings.

I'm sorry if I wasn't clear enough, but where **exactly** do the **multiple line-endings** come from? After downloading the master branch as zip file (i.e. without any conversions on client-side) and then looking at `nuklear.h` and files under `src/`, I can't find any inconsistency:
```
0$ for f in src/*; do python2 -c 'import sys; assert "\r" not in sys.stdin' < "$f"; done
0$ python2 -c 'import sys; assert "\r" not in sys.stdin' < nuklear.h 
0$ python2 -c 'import sys; assert "\n" not in sys.stdin' < nuklear.h 
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AssertionError
```
(ran on my Linux machine)

Reply:
The problem here seems to be, that when you clone on windows, git gives you all the files with `\r\n` line endings.
Then, `build.py` is run. The `print` statements outputs `\n` but the files loaded from disk, processed and outputted all contains `\r\n`, so now the generated `nuklear.h` has a mix of line endings.

I wish we could live in this world:
```
* text eol=lf
```

Reply:
@Hejsil sounds plausible - what about just replacing `.replace( '\r', '' )` on all opened files with `wb` (i.e. binary mode). That'd do, right?

Reply:
I think that could work. As long as this file is generated with one consistent line ending then I think git will be happy.

Reply:
> I'm sorry if I wasn't clear enough, but where **exactly** do the **multiple line-endings** come from?

> `print()` on Windows in Python 2.x (or at least in 2.7) appends \n to the end of the string you pass it. Git on Windows converts all line-endings to \r\n when you checkout, at least by default.

Hejsil explained it a bit clearer than I did, it seems.

> what about just replacing .replace( '\r', '' ) on all opened files with wb (i.e. binary mode). That'd do, right?

I'm not sure we should start doing this kind of thing, but if it works, it works. I still feel that simply requiring that the script be run with Py3 would be a better decision.

Reply:
I do agree we can restrict users to Python 3, but I still like the explicit replace (with a comment explaining "why") because that makes it fully explicit.

Reply:
Maybe this should be a separate pull request? Then we could get this one merged ;)